src/compiler/checker.ts

```diff
@@ -1526,17 +1534,53 @@ module ts {
                     }
                 }
 
+                function writeSymbolTypeReference(symbol: Symbol, typeArguments: Type[], pos: number, end: number) {
+                    if (!isReservedMemberName(symbol.name)) {

```

Comment:
Why do you need to check this? What are the cases where you hit a reserved member name that has type parameters, causing you to need to print it?